test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,285 @@
+// -----------------------------------------------------------------------------------------------------
+
+#include <benchmark/benchmark.h>
+
+#include <memory>
+#include <random>
+#include <utility>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna4.hpp>
+#include <seqan3/search/algorithm/all.hpp>
+#include <seqan3/test/performance/sequence_generator.hpp>
+#include <stdlib.h>
+
+using namespace seqan3;
+using namespace seqan3::test;
+
+struct Options
+{
+    size_t const sequence_length;
+    bool const repeating;
+    size_t const number_of_reads;
+    size_t const read_length;
+    double const prob_insertion;
+    double const prob_deletion;
+    double const simulated_errors;
+    uint8_t const searched_errors;
+    uint8_t const strata;
+    int repeats = 20;
+};
+
+template <Alphabet alphabet_t>
+void mutate_insertion(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<uint8_t> dis_alpha(0, alphabet_size<alphabet_t> - 1);
+    std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(seq) - overlap);
+    alphabet_t cbase;
+    seq.insert(seq.begin() + random_pos(gen), alphabet_t{}.assign_rank(dis_alpha(gen)));
+}
+
+template <Alphabet alphabet_t>
+void mutate_deletion(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(seq) - overlap);
+    seq.erase(seq.begin() + random_pos(gen));
+}
+
+template <Alphabet alphabet_t>
+void mutate_substitution(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<uint8_t> dis_alpha_short(0, alphabet_size<alphabet_t> - 2);
+    std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(seq) - overlap);
+    alphabet_t & cbase = seq[random_pos(gen)];
+    uint8_t crank = to_rank(cbase);
+    uint8_t rrank = dis_alpha_short(gen);
+    if (rrank >=  crank)
+        ++rrank;
+    cbase.assign_rank(rrank);
+}
+
+template <Alphabet alphabet_t>
+auto generate_reads(std::vector<alphabet_t> & ref,
+                    size_t const number_of_reads,
+                    size_t const read_length,
+                    float const simulated_errors_v,
+                    float const prob_insertion,
+                    float const prob_deletion,
+                    bool const error_dis = false,
+                    size_t const seed = 0)
+{
+    std::vector<std::vector<alphabet_t> > reads;
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<size_t> seeds (0, SIZE_MAX);
+    std::normal_distribution<> dis_error_count{0, simulated_errors_v};
+    std::uniform_real_distribution<double> probability(0.0, 1.0);
+
+    for (size_t i = 0; i < number_of_reads; ++i)
+    {
+        // TODO avoid mutating the same position multiple times
+        int simulated_errors;
```

Comment:
```suggestion
        uint8_t simulated_errors = error_dis ? simulated_errors_ : abs(std::round(dis_error_count(gen)));
```
and then remove the following if/else